vignettes/checks/plausibleAfterBirth.Rmd

```diff
@@ -0,0 +1,94 @@
+---
+title: "plausibleAfterBirth"
+author: "Maxim Moinat, Katy Sadowski"
+date: "`r Sys.Date()`"
+output:
+  html_document:
+    number_sections: yes
+    toc: yes
+---
+
+## Summary
+
+**Level**: Field check\
+**Context**: Verification\
+**Category**: Plausibility\
+**Subcategory**: Temporal\
+**Severity**: Characteristic
+
+## Description
+The number and percent of records with a date value in the **cdmFieldName** field of the **cdmTableName** table that occurs prior to birth.
+
+## Definition
+This check verifies that events happen after birth. This check is only run on fields where the **PLAUSIBLE_AFTER_BIRTH** parameter is set to **Yes**. The birthdate is taken from the `person table`, either the `birth_datetime` or composed from `year_of_birth`, `month_of_birth`, `day_of_birth` (taking 1st month/1st day if missing).
+
+- *Numerator*: The number of records with a non-null date value that happen prior to birth.
+- *Denominator*: The total number of records in the table with a non-null date value.
+- *Related CDM Convention(s)*: -Not linked to a convention-
+- *CDM Fields/Tables*: By default, this check runs on all date and datetime fields
+- *Default Threshold Value*: 1
+
+## User Guidance
+There might be valid reasons why a record has a date value that occurs prior to birth. For example, prenatal observations might be captured or procedures on the mother might be added to the file of the child. Therefore, some failing records are expected and the default threshold of 1% accounts for that.
+
+However, if more records violate this check, there might be an issue with incorrect birthdates or events with a default date. It is recommended to investigate the records that fail this check to determine the cause of the error and set proper dates. If it is impossible to fix, then implement one of these:
+
+ - Aggressive: Remove all patients who have at least one record before birth (if the birthdate of this patient is unreliable)
+ - Less aggressive: Remove all rows that happen before birth. Probably this should be chosen as a conventional approach for data clean up (if the event dates are unreliable)
+ - Conservative: set event date to birth date.
```

Comment:
maybe should also add the option of leaving as-is, i.e. for prenatal or other observations where the date is actually valid.

Reply:
I actually like that better for the conservative strategy. Modifying the event date might create other issues. Either it is correct and you leave it as is, or you don't trust the record and remove it, or you don't trust the person and remove it.